### PR TITLE
Adding state property - fixes #22

### LINKED ISSUE #22: There should be an interface that displays all things that aren't complete

### PR DESCRIPTION
**Pull request by escowles**

### CODE REVIEW

File: app/controllers/catalog_controller.rb [hunk 1]
```diff
@@ -234,4 +238,8 @@ class CatalogController < ApplicationController
     # mean") suggestion is offered.
     config.spell_max = 5
   end
+  def admin_state_facet
+    return unless current_user && (current_user.curation_concern_creator? || current_user.admin?)
```
Comment (tpendragon) [inline app/controllers/catalog_controller.rb]: Can we move this to ability.rb and use cancan?

File: app/models/concerns/common_metadata.rb [hunk 2]
```diff
@@ -15,6 +15,11 @@ module CommonMetadata
     property :source_metadata,
              predicate: ::RDF::URI.new('http://library.princeton.edu/terms/source_metadata'),
              multiple: false
+    property :state,
+             predicate: ::RDF::URI.new('http://fedora.info/definitions/1/0/access/objState'),
+             multiple: false do |index|
```
Comment (tpendragon) [inline app/models/concerns/common_metadata.rb]: Indenting is a little odd here, surprised rubocop ignored it.

File: app/presenters/curation_concerns_show_presenter.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+class CurationConcernsShowPresenter < CurationConcerns::WorkShowPresenter
+  delegate :date_created, :viewing_hint, :viewing_direction, :state, to: :solr_document
+
+  def state_badge
+    StateBadge.new(solr_document).render
```
Comment (tpendragon) [inline app/presenters/curation_concerns_show_presenter.rb]: Gorgeous

File: app/models/concerns/common_metadata.rb (hunk 2 above)
Comment (escowles) [inline app/models/concerns/common_metadata.rb]: Yeah, the previous property definitions are indented so their arguments are in a block (rubocop seems to recognize and allow this), but then the block must be indented two spaces.

Related, maybe we should define some RDF vocabs for these predicates, and then the property definitions might be shorter enough to fit on one line?

Comment (tpendragon) [inline app/models/concerns/common_metadata.rb]: > maybe we should define some RDF vocabs for these predicates

:+1: Is the fedora one not defined somewhere up the stack already?

File: app/controllers/catalog_controller.rb (hunk 1 above)
Comment (tpendragon) [inline app/controllers/catalog_controller.rb]: Thinking something like

```ruby
if can? :manage_state, SolrDocument # or some other model...
  ...
end
```

Comment (tpendragon) [inline app/controllers/catalog_controller.rb]: That should let you remove the Devise/current_user stuff added in this PR too.

File: app/validators/state_validator.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+class StateValidator < ActiveModel::Validator
+  def validate(record)
```
Comment (tpendragon) [inline app/validators/state_validator.rb]: Can we use an inclusivity validator for this? The error messages tend to be pretty. Something similar to how ViewingHintValidator works.

File: app/views/curation_concerns/base/_form_state.html.erb [hunk 5]
```diff
@@ -0,0 +1,13 @@
+<fieldset id="set-state">
+  <legend>State</legend>
+  <div class="form-group">
+    <label class="radio">
```
Comment (tpendragon) [inline app/views/curation_concerns/base/_form_state.html.erb]: Can we use simple form builders for these? Might push off some of the view branches

File: app/controllers/catalog_controller.rb (hunk 1 above)
Comment (escowles) [inline app/controllers/catalog_controller.rb]: Strangely, `can? :edit, ScannedResource` returns true for admin, curation_concern_creator and anonymous users.

Comment (tpendragon) [inline app/controllers/catalog_controller.rb]: Yeah, you need an instance of ScannedResource to check against for edit.

Comment (escowles) [inline app/controllers/catalog_controller.rb]: Hmm, different problem with `can? :edit, ScannedResource.new`: works for admin/curation_concern_creator, but it results in a 404 for anonymous users.

Comment (tpendragon) [inline app/controllers/catalog_controller.rb]: Maybe can? :create, ScannedResource ?

Comment (tpendragon) [inline app/controllers/catalog_controller.rb]: We may want an explicit permission for this - who can review may be different than who can create

Comment (escowles) [inline app/controllers/catalog_controller.rb]: `can? :create, ScannedResource` works, so I think we should go with that.  We can create a new permission for review later on if creators and reviewers really are separate.

File: app/models/concerns/common_metadata.rb (hunk 2 above)
Comment (escowles) [inline app/models/concerns/common_metadata.rb]: The Fedora 4 ontology is in RDF.rb, but not the Fedora 3 ontologies.  I'll add stub vocabs for it and the PUL/OpaqueNamespace terms.

**Pull request merged**